reflex/experimental/assets.py

```diff
@@ -1,57 +1,90 @@
+    rx.image(src=rx._x.asset(path="local_image.png"))
     ```
 
     Args:
-        relative_filename: The relative filename of the asset.
-        subfolder: The directory to place the asset in.
+        path: The relative path of the asset.
+        subfolder: The directory to place the shared asset in.
+        shared: Whether to expose the asset to other apps.
 
     Raises:
         FileNotFoundError: If the file does not exist.
+        ValueError: If subfolder is provided for local assets.
 
     Returns:
-        The relative URL to the copied asset.
+        The relative URL to the asset.
     """
+    assets = constants.Dirs.APP_ASSETS
+    backend_only = os.environ.get(constants.ENV_BACKEND_ONLY)
+
+    # Local asset handling
+    if not shared:
+        cwd = Path.cwd()
+        src_file_local = cwd / assets / path
+        if subfolder is not None:
+            raise ValueError("Subfolder is not supported for local assets.")
+        if not backend_only and not src_file_local.exists():
+            raise FileNotFoundError(f"File not found: {src_file_local}")
+        return f"/{path}"
+
+    # Shared asset handling
     # Determine the file by which the asset is exposed.
     calling_file = inspect.stack()[1].filename
     module = inspect.getmodule(inspect.stack()[1][0])
     assert module is not None
-    caller_module_path = module.__name__.replace(".", "/")
-
-    subfolder = f"{caller_module_path}/{subfolder}" if subfolder else caller_module_path
-
-    src_file = Path(calling_file).parent / relative_filename
 
-    assets = constants.Dirs.APP_ASSETS
     external = constants.Dirs.EXTERNAL_APP_ASSETS
+    src_file_shared = Path(calling_file).parent / path
+    if not src_file_shared.exists():
+        raise FileNotFoundError(f"File not found: {src_file_shared}")
 
-    if not src_file.exists():
-        raise FileNotFoundError(f"File not found: {src_file}")
+    caller_module_path = module.__name__.replace(".", "/")
+    subfolder = f"{caller_module_path}/{subfolder}" if subfolder else caller_module_path
 
-    # Create the asset folder in the currently compiling app.
-    asset_folder = Path.cwd() / assets / external / subfolder
-    asset_folder.mkdir(parents=True, exist_ok=True)
+    # Symlink the asset to the app's external assets directory if running frontend.
+    if not backend_only:
+        # Create the asset folder in the currently compiling app.
+        asset_folder = Path.cwd() / assets / external / subfolder
+        asset_folder.mkdir(parents=True, exist_ok=True)
 
-    dst_file = asset_folder / relative_filename
+        dst_file = asset_folder / path
 
-    if not dst_file.exists():
-        dst_file.symlink_to(src_file)
+        if not dst_file.exists() and (
+            not dst_file.is_symlink() or dst_file.resolve() != src_file_shared.resolve()
+        ):
+            dst_file.symlink_to(src_file_shared)
```

Comment:
this makes me a bit nervous because symlinks are not generally enabled or available to unprivileged users on windows.

can we have a fallback path that just copies the files on windows or if this operation fails

Reply:
this PR did not introduce the symlinks, they were there before. i can address it in another PR